OpenRA.Mods.Common/HitShapes/Circle.cs

```diff
@@ -48,25 +51,27 @@ public WDist DistanceFromEdge(WVec v)
 
 		public WDist DistanceFromEdge(WPos pos, Actor actor)
 		{
-			var actorPos = actor.CenterPosition;
+			var orientation = actor.Orientation;
+			var actorPos = actor.CenterPosition + LocalOffset.Rotate(orientation);
 
 			if (pos.Z > actorPos.Z + VerticalTopOffset)
-				return DistanceFromEdge(pos - (actorPos + new WVec(0, 0, VerticalTopOffset)));
+				return DistanceFromEdge((pos - (actorPos + new WVec(0, 0, VerticalTopOffset))).Rotate(-orientation));
```

Comment:
I don't think this is necessary for the current state of this PR, so if that sounds too scary then it might be more efficient to drop this commit for now.